### PR TITLE
Stop signs integration

### PR DESCRIPTION
**Pull request by phucla0122**
Stop sign is now detected, but not very stable
The closer the car is to stop sign, the easier for it to detect
Limitation, if the car wants to detect stop sign from a far, it could mistakenly detect something else as a stop sign

### CODE REVIEW

File: src/sign_car_recognition/scripts/stop_sign.xml
Comment (nicolastuttle) [inline src/sign_car_recognition/scripts/stop_sign.xml:1]: What is this? We don't have other xml files for car detection

**Review by robellgabriel — COMMENTED** [review]
Does the car stop when detecting stop signs then goes? Or does it only just detect the stop sign at the moment

File: src/sign_car_recognition/scripts/stop_sign.xml
Comment (phucla0122) [reply src/sign_car_recognition/scripts/stop_sign.xml:1]: That's only for sign detection, for car, we use yolov5s model to detect

**Comment by phucla0122** [comment]
> Does the car stop when detecting stop signs then goes? Or does it only just detect the stop sign at the moment

It just detects the sign for now, however, it's not a guarantee detection. The detection rate has to be fine tune quite a bit more to be able to guarantee detect the sign

File: src/central_control/scripts/central_control.py [hunk 1]
```diff
@@ -432,13 +440,35 @@ def handle_object_recognition(self, res: DetectionResults):
...
+    '''def handle_stop_sign(self, res: DetectionResults):
+        if not self.ready:
+            self.ready = True
+        # Sort through and deal with sign detection
+        self.sign_data.clear()
+        detection_list: List[DetectionResult] = res.detection_results
+        for detection in detection_list:
+            if detection.class_num == 11 and detection.depth < 50 and detection.confidence > 0.2:
+                # Stop sign. Depth is likely going to be smaller than 25 because of the image resolution
+                self.sign_data.append(detection)'''
```
Comment (robellgabriel) [inline src/central_control/scripts/central_control.py]: you can delete this as the method above is pretty much identical

**Review by robellgabriel — APPROVED** [review]
Great work! One step closer to the City environment

File: src/central_control/scripts/central_control.py [hunk 2]
```diff
@@ -193,6 +195,10 @@ def listen(self):
                     # Heuristic values
                     if sign.depth < REACT_DEPTH:
                         self.stop_state = StopState.STOPPING
+                        self.car_controls.throttle = STOP_THROTTLE
+                        self.car_controls.brake = HOLD_BRAKE
+                        #self.bridge.set_controls(self.car_controls)
+                        print("A")
```
Comment (nicolastuttle) [inline src/central_control/scripts/central_control.py]: Do you still need this debug print?

File: src/central_control/scripts/central_control.py (hunk 1 above)
Comment (phucla0122) [reply src/central_control/scripts/central_control.py]: Fair enough, gonna do that

File: src/central_control/scripts/central_control.py (hunk 2 above)
Comment (phucla0122) [reply src/central_control/scripts/central_control.py]: I'll remove that

**Review by nicolastuttle — APPROVED** [review]

**Pull request merged**